library/ssl_srv.c

```diff
@@ -3585,15 +3585,19 @@ static int ssl_parse_certificate_verify( mbedtls_ssl_context *ssl )
     /* Read the message without adding it to the checksum */
     do {
 
-        if( ( ret = mbedtls_ssl_read_record_layer( ssl ) ) != 0 )
+        do ret = mbedtls_ssl_read_record_layer( ssl );
+        while( ret == MBEDTLS_ERR_SSL_CONTINUE_PROCESSING );
```

Comment:
@RonEld The quote suggests that `return( MBEDTLS_ERR_SSL_CONTINUE_PROCESSING )` is within the failure branch of `mbedtls_ssl_fetch_input`, but it is within the failure branch for `mbedtls_ssl_parse_record`, which is only executed if the interface to the underlying transport `mbedtls_ssl_fetch_input` was successful. In case the latter fails, the error is escalated back to the user, and no looping takes place. 

The question of whether looping on unexpected records is desirable or not is a separate issue that is not touched upon by this PR (both the old and the suggested code do looping in some way) and is open to discussion - see IOTSSL-1424.